src/test/java/org/plp/implementations/plpisa/PlpFileTest.java

```diff
@@ -0,0 +1,29 @@
+package org.plp.implementations.plpisa;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class PlpFileTest {
+
+    @Mock PlpProgram mockProgram;
+    @Mock PlpFile mockExistingPlpFile;
+
+    @Test
+    void createPlpFileSuccessFulTest() {
+        Assertions.assertDoesNotThrow(() -> new PlpFile("test.asm", mockProgram));
+    }
+
+    @Test
+    void createPlpFileFailureTest() {
+        when(mockProgram.getAsmFile("test.asm")).thenReturn(mockExistingPlpFile);
```

Comment:
mockProgram.validateFileName instead.

Reply:
As we are removing the AsmProgram reference from the AsmFile, this will not be needed.